src/awkward/cpp-headers/awkward/GrowableBuffer.h

```diff
@@ -132,7 +141,10 @@ namespace awkward {
     ///
     /// Changes the data type from `PRIMITIVE` to `TO_PRIMITIVE`/
     template <typename TO_PRIMITIVE>
-    void
+    typename std::enable_if<(!awkward::is_tt<std::complex, TO_PRIMITIVE>::value &&
+                             !awkward::is_tt<std::complex, PRIMITIVE>::value) ||
+                            (awkward::is_tt<std::complex, TO_PRIMITIVE>::value &&
+                             awkward::is_tt<std::complex, PRIMITIVE>::value)>::type
```

Comment:
This template specialization handles any conversion between the arithmetic types or between the complex types. The latter is tested for all combinations of the type of the real and imaginary components: `float`, `double`, and `long double`.

Reply:
Okay! I see the tests, and that demonstrates that it's working.